### PR TITLE
R1 final review

### PR DESCRIPTION
**Pull request by romanmandryk**
(removed checklist, see below)

### CODE REVIEW

**Comment by GoodluckH** [comment]
## Documents
- [ ] ICD
  - [ ] Current interfaces (grpc & rest)
  - [ ] Current relevant document links
- [ ] SDD
  - [ ] Description of grpc & rest handler implementations
  - [ ] Current sequence diagrams where applicable
  - [ ] Current relevant document links
- [ ] CONOPS
  - [ ] Current information
  - [ ] Bibliography (if any citations)
- [ ] README (see svc-cargo for example)
  - [ ] Badges
  - [ ] Accurate installation instructions
  - [ ] Current links to other documents
  - [ ] Arrow DAO links

## Code (not applicable to examples/ folder)
- [ ] No unhandled unwrap()s
- [ ] Accurate (!) code comments on all functions
- [ ] No `# Arguments` and `# Returns` in function comments
- [ ] Function arguments checked where possible
- [ ] Sufficient logger (info, warn, and debug) messages
- [ ] Sensible Types (unsigned vs. signed)

## REST API
- [ ] Return codes accurately documented with openapi
- [ ] Return codes make sense
- [ ] Sufficiently descriptive error messages

## Cargo.toml(s)
- [ ] Package versions correct
- [ ] No minor versions specified
- [ ] Links to crates.io package, not git branch

**Comment by GoodluckH** [comment]
Are you planning to include CONOPS for R2?

File: server/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    async fn is_ready(
+        &self,
+        _request: Request<ReadyRequest>,
+    ) -> Result<Response<ReadyResponse>, Status> {
+        let response = ReadyResponse { ready: true };
+        Ok(Response::new(response))
+    }
+}
+
+async fn init_router(mut vertiport_client: VertiportRpcClient<Channel>) {
```
Comment (GoodluckH) [inline server/src/main.rs:122]: Missing docs

File: server/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    info!("Initializing router with {} vertiports ", vertiports.len());
+    if !is_router_initialized() {
+        let res = init_router_from_vertiports(&vertiports);
+        if res.is_err() {
+            error!("Failed to initialize router: {}", res.err().unwrap());
+        }
+    }
+}
+
+async fn init_grpc_clients() {
```
Comment (GoodluckH) [inline server/src/main.rs:143]: Missing docs

File: server/src/queries.rs [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    SearchFilter, UpdateFlightPlan,
+};
+
+use tokio;
+use tonic::{Request, Response, Status};
+use uuid::Uuid;
+
+const CANCEL_FLIGHT_SECONDS: u64 = 30;
+
+fn unconfirmed_flight_plans() -> &'static Mutex<HashMap<String, FlightPlanData>> {
```
Comment (GoodluckH) [inline server/src/queries.rs:23]: Missing docs

File: server/src/queries.rs [hunk 4]
```diff
@@ -0,0 +1,292 @@
+use crate::scheduler_grpc::{
```
Comment (GoodluckH) [inline server/src/queries.rs:1]: Missing file level docs

File: server/src/queries.rs [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+use uuid::Uuid;
+
+const CANCEL_FLIGHT_SECONDS: u64 = 30;
+
+fn unconfirmed_flight_plans() -> &'static Mutex<HashMap<String, FlightPlanData>> {
+    static INSTANCE: OnceCell<Mutex<HashMap<String, FlightPlanData>>> = OnceCell::new();
+    INSTANCE.get_or_init(|| Mutex::new(HashMap::new()))
+}
+
+fn cancel_flight_after_timeout(id: String) {
```
Comment (GoodluckH) [inline server/src/queries.rs:28]: Missing docs

**Comment by romanmandryk** [comment]
Conops are done by AM - see this PR - it will be part of R2

File: server/src/queries.rs [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+        flights: [item].to_vec(),
+    };
+    info!(
+        "query_flight returning: {} flight plans",
+        &response.flights.len()
+    );
+    Ok(Response::new(response))
+}
+
+fn get_fp_by_id(id: String) -> Option<FlightPlanData> {
```
Comment (GoodluckH) [inline server/src/queries.rs:174]: Missing docs

File: server/src/queries.rs [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+    unconfirmed_flight_plans()
+        .lock()
+        .expect("Mutex Lock Error getting flight plan from temp storage")
+        .get(&id)
+        .cloned()
+}
+
+fn remove_fp_by_id(id: String) -> bool {
```
Comment (GoodluckH) [inline server/src/queries.rs:182]: Missing docs

File: server/src/main.rs [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+    let vertiports = vertiport_client
+        .vertiports(Request::new(SearchFilter {
+            search_field: "".to_string(),
+            search_value: "".to_string(),
+            page_number: 0,
+            results_per_page: 50,
+        }))
+        .await
+        .unwrap()
+        .into_inner()
```
Comment (GoodluckH) [inline server/src/main.rs:132]: how's this unwrap handled if it's none or err?

File: client-grpc/Cargo.toml [hunk 9]
```diff
@@ -0,0 +1,23 @@
+[package]
+edition = "2021"
+name    = "svc-scheduler-client-grpc"
+version = "0.2.5-develop.3"
```
Comment (owlot) [inline client-grpc/Cargo.toml:4]: Please reset the version to "0.1.0" so the release workflow won't run into issues and we get a proper CHANGELOG generated

File: server/Cargo.toml [hunk 10]
```diff
@@ -0,0 +1,44 @@
+[package]
+edition = "2021"            # Allowed values are 2015, 2018, 2021
+name    = "svc-scheduler"
+version = "0.2.5-develop.3"
```
Comment (owlot) [inline server/Cargo.toml:4]: Please reset the version to "0.1.0" so the release workflow won't run into issues and we get a proper CHANGELOG generated

**Review by owlot — CHANGES_REQUESTED** [review]

File: server/src/main.rs (hunk 8 above)
Comment (romanmandryk) [reply server/src/main.rs:132]: You're right - this was unhandled. Fixed now.

**Review by GoodluckH — APPROVED** [review]

**Review by owlot — APPROVED** [review]

**Review by GoodluckH — APPROVED** [review]

**Review by GoodluckH — APPROVED** [review]

**Pull request merged**